lib/modules/manager/pip-compile/extract.ts

```diff
@@ -0,0 +1,55 @@
+import { logger } from '../../../logger';
+import { extractPackageFile as extractRequirementsFile } from '../pip_requirements/extract';
+// TODO(not7cd): enable in the next PR, when this can be properly tested
+// import { extractPackageFile as extractSetupPyFile } from '../pip_setup';
+// import { extractPackageFile as extractSetupCfgFile } from '../setup-cfg';
+import type { ExtractConfig, PackageFileContent } from '../types';
+import type { SupportedManagers } from './types';
+
+function matchManager(filename: string): SupportedManagers | 'unknown' {
+  if (filename.endsWith('setup.py')) {
+    return 'pip_setup';
+  }
+  if (filename.endsWith('setup.cfg')) {
+    return 'setup-cfg';
+  }
+  if (filename.endsWith('pyproject.toml')) {
+    return 'pep621';
+  }
+  // naive, could be improved, maybe use pip_requirements.fileMatch
+  if (filename.endsWith('.in')) {
```

Comment:
🤔 it's a kind of breaking when users didn't use `.in` files already as input? can we treat unknown as warning and fallback to requirements for now?

Reply:
let's fallback